lib/pharos/cluster_manager.rb

```diff
@@ -42,25 +42,28 @@ def validate
 
     def apply_phases
       apply_phase(Phases::ValidateHost, config.hosts, ssh: true, parallel: true)
-      apply_phase(Phases::MigrateMaster, config.master_hosts, ssh: true, parallel: true)
       apply_phase(Phases::ConfigureHost, config.hosts, ssh: true, parallel: true)
+
+      apply_phase(Phases::MigrateMaster, config.master_hosts, ssh: true, parallel: true)
       apply_phase(Phases::ConfigureCfssl, config.etcd_hosts, ssh: true, parallel: true)
       apply_phase(Phases::ConfigureEtcdCa, config.etcd_hosts[0...1], ssh: true, parallel: false)
```

Comment:
Yes, this is intentional trickery from  #276... if the `etcd_hosts` is empty, then this is an empty array and the phase is skipped; otherwise it's equivalent to `[config.etcd_hosts.first]`. This should really be better written out more explicitly.